src/main/java/com/mathworks/ci/BuildArtifactAction.java

```diff
@@ -0,0 +1,208 @@
+package com.mathworks.ci;
+
+import hudson.FilePath;
+import hudson.model.Action;
+import hudson.model.Run;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.CheckForNull;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+
+public class BuildArtifactAction implements Action {
+    private Run<?, ?> build;
+    private FilePath workspace;
+    private int totalcount;
+    private int skipcount;
+    private int failCount;
+    private static String ROOT_ELEMENT = "taskDetails";
+    private static String BUILD_ARTIFACT_FILE = "/buildArtifact.json";
+
+    public BuildArtifactAction(Run<?, ?> build, FilePath workspace) {
+        this.build = build;
+        this.workspace = workspace;
+    }
+
+    public BuildArtifactAction() {
+    }
```

Comment:
It doesn't look like this version of the constructor is ever called. Maybe we can remove it?

Reply:
Done.